src/disjoint_mut.rs

```diff
@@ -496,12 +582,6 @@ impl Bounds {
         let b = &other.range;
         a.start < b.end && b.start < a.end
     }
-
-    fn multiply(self, multiple: usize) -> Bounds {
-        let start = self.range.start * multiple;
-        let end = self.range.end * multiple;
```

Comment:
Why not just check range.end here and only multiply if not `usize::MAX`? Would be a lot simpler than all the other stuff here.

Reply:
Using `usize::MAX` also prohibits other optimizations since it conflates actual `usize::MAX`.  And this isn't complex code, just somewhat repetitive.  I can remove the division ones since I ended up not needing them, though.  I can try to see if I can rearrange things to only need `fn mul` on `Range<usize>`, but I don't think this code is very complex.

Reply:
I removed `fn div` in [`edf6ae0..36164b9`](https://github.com/memorysafety/rav1d/compare/edf6ae0621aaab9438635a2a5a1f8ac03af21a8e..36164b9dbf24c725b22bf3b9a54df86d05cd7978).  I can see about other simplifications, too, but I'd like to merge this as is and fix changes on top of the stack, since this PR is at the bottom.  Also, some extra lines of code here may be worth the performance, since `usize::MAX` checks may be in the hot path of whatever indexing we're doing.